openwisp_firmware_upgrader/tests/test_api.py

```diff
@@ -568,6 +568,40 @@ def test_category_list_filter_org_admin(self):
             r = self.client.get(url, data_filter)
         self.assertEqual(r.data['results'], serialized_list)
 
+    def test_category_serializer_validation(self):
+        url = reverse('upgrader:api_category_list')
+
+        with self.subTest('Test non-superuser cannot create shared category'):
+            data = {'name': 'Shared Category', 'organization': ''}
+            r = self.client.post(url, data)
+            self.assertEqual(r.status_code, 400)
+            self.assertEqual(
+                r.data['organization'][0],
+                'Only superusers can create or edit shared categories',
+            )
```

Comment:
Isn't this redundant with https://github.com/openwisp/openwisp-firmware-upgrader/pull/293/files#diff-55c81ae6cc3ae3043324b170be902f6c1476ecb63469dddbf53018fc79e9926dR483-R487 ?

Are you testing speecific cases which weren't tested in the code added by @pandafy in https://github.com/openwisp/openwisp-firmware-upgrader/pull/293/files#diff-55c81ae6cc3ae3043324b170be902f6c1476ecb63469dddbf53018fc79e9926dR446-R500?
If I am right and these tests are redundant, please remove them. Next time if a mentor asks you to add a test but you see that the test they're asking you to add would be redundant, let them know.